src/BlockTypeRegistry.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+
+
+
+
+#include <map>
+
+
+
+
+
+// fwd:
+class BlockState;
+
+
+
+
+
+/** Complete information about a single block type.
+The BlockTypeRegistry uses this structure to store the registered information. */
+class BlockInfo
+{
+public:
+
+	/** Callback is used to query block hints dynamically, based on the current BlockState.
+	Useful for example for redstone lamps that can be turned on or off. */
+	using HintCallback = std::function<AString(const AString & aTypeName, const BlockState & aBlockState)>;
```

Comment:
Perhaps we should also provide the world and vector of the requested block?

Reply:
Not really - that would mean that the hint would need to be queried for each block in the world separately, while right now it means that the hint can be queried once per BlockArea's palette.